test/jruby/test_zlib.rb

```diff
@@ -517,6 +517,24 @@ def test_error_input
     assert_equal("not in gzip format", e.message)
     assert_equal("foobarzothoge", e.input)
   end
+
+  def test_gzip_reader_ungetc
```

Comment:
If these don't exist in either MRI's suite or ruby/spec, it would be nice to add them there. We usually just put JRuby-specific stuff in test/jruby.